File: application/src/Caravelo/Bundle/ApiBundle/Model/QuestionDto.php
<?php

namespace Caravelo\Bundle\ApiBundle\Model;

use Caravelo\Modules\Survey\Domain\Model\Question;

class QuestionDto
{
    public $id;

    public $title;

    public $createdAt;

    public $answers = [];

    public function parseQuestion(Question $question)
    {
        $this->id = $question->getId();
        $this->title = $question->getTitle();
        $this->createdAt = $question->getCreatedAt()->format('Y-M-D\TH:I');


        foreach($question->getAnswers() as $answer)
        {
            $answerDto = new AnswerDto();
            $answerDto->parseAnswer($answer);
            $this->answers[] = $answerDto;
        }
    }
}

File: application-java/src/main/java/iruiz/app/dto/QuestionDto.java
package iruiz.app.dto;

import java.util.ArrayList;

public class QuestionDto {

    public String title;

    public String createdAt;

    public ArrayList<AnswerDto> answers;
    
    public QuestionDto(){
    	this.answers = new ArrayList<AnswerDto>();
    }
    
    public void addAnswer(AnswerDto answer){
    	this.answers.add(answer);
    }
}
